### DISCUSSION TITLE
cannot use svcg

### DISCUSSION THREAD

**Discussion by kevinrex66** [voicepaw/so-vits-svc-fork#691] [Q&A]
**Describe the bug**
The conda enviroment is OK and i infer my own model what i trained before successfully, but i cannot run the svcg 

Traceback (most recent call last):
  File "/home/anaconda3/envs/so-vits-svc-fork/bin/svcg", line 5, in <module>
    from so_vits_svc_fork.gui import main
  File "/home/anaconda3/envs/so-vits-svc-fork/lib/python3.10/site-packages/so_vits_svc_fork/gui.py", line 10, in <module>
    import sounddevice as sd
  File "/home/anaconda3/envs/so-vits-svc-fork/lib/python3.10/site-packages/sounddevice.py", line 71, in <module>
    raise OSError('PortAudio library not found')
OSError: PortAudio library not found

PS：The infer result sang out of the tune, how can i improve the result except add dataset?

**Comment by Agamemnoid** [comment]
sudo apt-get install libportaudio2

Let me know if that works.

**Comment by kevinrex66** [comment]
> sudo apt-get install libportaudio2
> 
> Let me know if that works.

It works,but how can i improve the model, it sang out of the tune?

**Comment by Agamemnoid** [comment]
1:
I personally run the command
svc pre-hubert -fm crepe

BEFORE training the model. Then during inference I use the crepe auto-predict method.

2:
Train longer. Depending on your dataset. I personally still see improvements up to 1 Million steps. It's definitely diminishing returns, but it does exist.

3:
Better quality target. Not your training dataset, but the song you're inferencing onto. Try another one, clearer, make sure you use vocals only with absolutely no background noise. Make sure it's as clear as possible.

Other than that, you're out of options. It's quite possible that with a shitty dataset you will simply never ever have a good model.

**Comment by upColor** [comment]
@Agamemnoid  Thank you! Now I know how to change the F0 prediction method before training locally.

Before that, I only know how to change F0 method on the GUI for the inference, and I find that "parselmouth" gives the best inference result in my case.

> I personally still see improvements up to 1 Million steps

You mean after 1 million steps, the loss function shown on the TensorBoard is still decreasing? 
That is surprising! During my training, after training for 3 hours with batch_size = 5, number of sample files = 800, and 160 steps for a single epoch, the loss function is no longer decreasing after 90 epochs, which is 90 X 800 / 5 =  14400 steps.
